Source/WebCore/platform/mediastream/gstreamer/GStreamerVideoCapturer.cpp

```diff
@@ -39,12 +39,110 @@ GStreamerVideoCapturer::GStreamerVideoCapturer(const char* sourceFactory)
 GstElement* GStreamerVideoCapturer::createConverter()
 {
     auto converter = gst_parse_bin_from_description("videoscale ! videoconvert ! videorate", TRUE, nullptr);
+#if USE(WESTEROS_SINK)
+// If camera supports only x-raw, westerossink expects encoded form. So pass through an h264 encoder and parser after probing platform.
+// If camera supports encoded formats, use a dummy identity element to skip converting to x-raw.
+    const char* codec = getPreferredCodec().c_str();
+    GST_INFO_OBJECT(m_pipeline.get(), "Preferred codec is %s ", codec);
+
+    if (strcmp (codec, "video/x-raw") == 0)
+    {
+        GstElementFactory *h264encoder = getEncoder ("video/x-h264");
```

Comment:
Maybe it would be nice to remove the whole ifdef and use this code in all the cases. But <strike>that would require a new branch here in case there's no H264 encoder available in the platform</strike>. In that case, the original `converter = gst_parse_bin_from_description("videoscale ! videoconvert ! videorate", TRUE, nullptr)` should be used.
Oh, I see that you already did this in the second commit of the PR. Just double-check that there are no leaks if `converter` is overwritten from its initial value in any of the branches. Manually unref the older value, or just use GRefPtr<GstElement> as the type of the variable for auto-unreffing. Access it with `.get()`.